### PR TITLE
Make MermaidFlowGenerator more resilient to class loading errors

### PR DESCRIPTION
**Pull request by tnleeuw**
I might have a test for this too but it requires changing the scanner to also allow scanning classes on the test classpath and I need to verify that there are no negative side-effects on other tests.

### CODE REVIEW

**Comment by tnleeuw** [comment]
@nielsm5 I added some tests in a separate commit. If you do not like those changes, it's best I have to adding a unit test for this case. We can revert that commit but keep the actual fixes. :-)

**Review by jacodg — APPROVED** [review]

File: core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java [hunk 1]
```diff
@@ -93,11 +93,17 @@ public void afterPropertiesSet() throws Exception {
...
+	private String compileFrankElementList() throws SAXException {
 		try (SaxDocumentBuilder builder = new SaxDocumentBuilder("root")) {
 			List<String> classNames = findAllFrankElements();
 			for(String className : classNames) {
-				Class<?> clazz = Class.forName(className);
+				Class<?> clazz;
+				try {
+					clazz = Class.forName(className);
+				} catch (ExceptionInInitializerError | NoClassDefFoundError | Exception e) {
+					log.info("Found class [{}] on classpath which cannot be loaded due to exception", className, e);
```
Comment (jkosternl) [inline core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:104]: This is not a serious message that should warn an user / integration specialist?

File: core/src/test/java/org/frankframework/configuration/digester/TestAnnotationUtils.java [hunk 2]
```diff
@@ -219,7 +219,10 @@ protected String buildDefaultBeanName(@Nonnull BeanDefinition definition) {
 		Set<Class<?>> interfazes = new HashSet<>();
 		String[] names = scanner.getRegistry().getBeanDefinitionNames();
 		for (String beanName : names) {
-			if(beanName.contains(this.getClass().getCanonicalName()) || beanName.startsWith("org.frankframework.credentialprovider")) continue; //Ignore this class
+			if(beanName.contains(this.getClass().getCanonicalName())
+					|| beanName.startsWith("org.frankframework.credentialprovider")
+					|| beanName.endsWith(".UnloadableClass")
+			) continue; //Ignore this class
```
Comment (jkosternl) [inline core/src/test/java/org/frankframework/configuration/digester/TestAnnotationUtils.java:225]:
```suggestion
			) continue; //Ignore this class because credentialProviders don't love class loading???
```
Perhaps a 'why' can be added?

**Review by jkosternl — APPROVED** [review]

File: core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java (hunk 1 above)
Comment (tnleeuw) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:104]: I don't think so. I think that if you know you do not have SAP-CD libraries on your classpath but you have that module loaded because it's just part of the bundle, you do not need to see that as a warning every time you start up the system.

Comment (jkosternl) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:104]: Ok thanks. I just didn't know how this is used in practice.

Comment (nielsm5) [reply core/src/main/java/org/frankframework/util/flow/MermaidFlowGenerator.java:104]: This is just for the flow visualization, in fact, only for the colors or Frank!Elements 😄

**Review by nielsm5 — APPROVED** [review]
